### ISSUE TITLE
qthelp: keywords entries are wrong

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#791]
Generating qthelp with index entries doesn't work with sphinx 1.1 

Whereas sphinx < 1.1 created for example

```
<keywords>
    <keyword name="eins" ref="tcl.html#index-0"/>
    <keyword name="zwei" ref="tcl.html#index-1"/>
</keywords> 
```

Sphinx >=1.1 creates

```
<keywords>
    <keyword name="eins" ref="('', 'tcl.html#index-0')"/>
    <keyword name="zwei" ref="('', 'tcl.html#index-1')"/>
</keywords>    
```

which results in non-functional links...

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/791
- Originally reported by: [Broder1977](https://bitbucket.org/Broder1977)
- Originally created at: 2011-10-13T16:40:28.439

**Comment by shimizukawa** [comment]
_From [Broder1977](https://bitbucket.org/Broder1977) on 2011-10-13 14:49:35+00:00_

I was able to fix this for my needs by changing line 266 in builders/qthelp.py as given below, but may be this needs more work... 

Greetings Björn

{{{
#!python

```
    # TEMPORARY FIX #
    ref_ = ref[-1]

    if id:
        item = ' '*12 + '<keyword name="%s" id="%s" ref="%s"/>' % (
                                                            name, id, ref_)
    else:
        item = ' '*12 + '<keyword name="%s" ref="%s"/>' % (name, ref_)  
```

}}}

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-11-01 08:26:46+00:00_

Closes #791: Fix QtHelp and HtmlHelp index entry links.

→ <<cset 8fdeaec0184e8ded18c09c12d0ed695d532843e5>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:33:11+00:00_

Removing version: 1.1 (automated comment)
